app/controllers/user_checklists_controller.rb

```diff
@@ -0,0 +1,195 @@
+end
+class MissingUserChecklistParameterError < UserChecklistControllerError
+end
+class UserChecklistNotFoundError < UserChecklistControllerError
+end
+
+#
+# Note that there are NO new, create, edit, update, or destroy actions.
+# A user cannot create a new checklist, nor can a user edit one or destroy one.
+# Nor can an admin create a new one or edit one manually or destroy one
+#
+#
+
+class UserChecklistsController < ApplicationController
+
+
+  def index
+    authorize_user_checklist_class
+
+    # Get all the checklists for the user in the path. Ex: If the current user is an admin,
+    # then we want to be sure to see the checklists for a specific user, not for the admin.
+    @user = current_user.admin? ? User.find(params[:user_id]) : current_user
+
+    found_lists = UserChecklist.where(user: @user) #.includes(:master_checklist)
+    @user_checklists = found_lists.blank? ? [] : found_lists
+  end
+
+
+  def show
+    set_user_checklist
+    authorize_user_checklist
+  end
+
+
+  def show_progress
+    @user = User.find(progress_params[:user_id])
+
+    @membership_guidelines = membership_guideline_root_user_checklist(@user)
+    @overall_progress = membership_guidelines_percent_complete(@user)
+
+    @user_checklist = UserChecklist.find(progress_params[:user_checklist_id])
+    authorize_user_checklist
+
+    if @overall_progress == 100
+      render :membership_guidelines_completed
+    end
+  end
+
+
+  # @return Array[<Hash<checklist_id: [Date]>>] - a list of user_checklists that had their date_completed changed.
+  #   Each item in the list is a simple Hash of { checklist_id: <id>, date_completed: <the updated date_completed>}
+  #   and a success or fail with error message.
+  #
+  # Don't render a new page.  Just update info as needed and send
+  # 'success' or 'fail' info back.
+  def all_changed_by_completion_toggle
+    raise 'Unsupported request' unless request.xhr?
+
+    authorize UserChecklist
+
+    changed_checklists = []
+
+    user_checklist_id = params[:user_checklist_id]
+    user_checklist = UserChecklist.find_by_id(user_checklist_id)
+
+    date_completed = nil # default
+    status = 'ok'
+    error_text = ''
+
+    if user_checklist
+      # list of all those changed_by date_completed toggled
+      # toggle the date_completed and give me the list of all those changed
+      # all_changed_by_completion_toggle
+      checklists_changed = user_checklist.all_changed_by_completion_toggle
+      changed_checklists.concat(checklists_changed.map { |checklist_changed| hash_id_date_completed(checklist_changed) })
+    else
+
+      status = 'error' # not found
+      error_text = t('.not_found', user_checklist_id: user_checklist_id)
+
+      #raise ActiveRecord::RecordNotFound,  "UserChecklist not found! user_checklist_id = #{user_checklist_id}"
+    end
+
+    render json: { status: status, changed_checklists: changed_checklists, date_completed: date_completed, error_text: error_text }
+  end
```

Comment:
Curious why you didn't use `handle_xhr_request` here?

Reply:
Just hadn't put it into place everywhere yet. :-)